deepspeed/runtime/engine.py

```diff
@@ -297,7 +297,8 @@ def __init__(
         self.timers = SynchronizedWallClockTimer()
         # Throughput timer
         self.tput_timer = ThroughputTimer(
-            batch_size=self.train_micro_batch_size_per_gpu(),
+            train_micro_batch_size_per_gpu=self.train_micro_batch_size_per_gpu(),
+            gradient_accumulation_steps=self.gradient_accumulation_steps(),
```

Comment:
We don't need timer to be aware of `gradient accumulation steps` so that client for example could report progress on micro steps if they wish.

Reply:
I this case, do we also want to use `batch_size=self.train_batch_size()` and don't even pass `num_workers`? It makes little sense to have it for the same reason, otherwise we need to accumulate samples I believe...

Reply:
I think we should keep `micro_batch_size` and `num_workers` since the timer.start()/stop() are invoked on micro batch size granularity. As I suggested in elsewhere, the client will inform timer of global batch size boundaries.

Reply:
In this case, should I also change pipe/engine [here](https://github.com/microsoft/DeepSpeed/blob/ef869377e91660a649d16d8f416841ed9b5dbed3/deepspeed/runtime/pipe/engine.py#L107) to use `self.train_micro_batch_size_per_gpu()` without gradient accumulation steps as well?

Reply:
But to your point, we could simplify as 
 ```python
      train_micro_batch_size=self.train_micro_batch_size_per_gpu() * self.dp_world_size
````

That will simplify timer code.

Reply:
Right, since you mentioned you want the caller to decide the reported batch size, rather than let the timer calculate it (in the spirit of `explicit is better than implicit`). I will make the change.